silverback-integration-kafka/src/Silverback.Integration.Kafka/Messaging/Broker/Kafka.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Confluent.Kafka;
+
+namespace Silverback.Messaging.Broker
+{
+    public class Kafka : Broker
+    {
+        private readonly ConcurrentBag<(string topic,
+            Dictionary<string, object> settings,
+            Func<IDeliveryResultHandler> handlerFactory,
+            IProducer producer)> _producers = new ConcurrentBag<(string, Dictionary<string, object>, Func<IDeliveryResultHandler>, IProducer)>();
+
+        /// <inheritdoc />
+        public override IConsumer GetConsumer(IEndpoint endpoint)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <inheritdoc />
+        public override IProducer GetProducer(IEndpoint endpoint)
+        {
+            var producerSettings = _producers.First(p => p.topic == endpoint.Name);
```

Comment:
But then, at the end...it's one producer per broker or one producer per endpoint?

Reply:
One producer per endpoint...

Reply:
Shouldn't it be one producer per broker (unless a different configuration is desired for a certain endpoint)? Wasn't it the point of our discussion?
As per the documentation you forwarded me "You should generally avoid creating multiple Consumer or Producer instances", right?

Reply:
> Shouldn't it be one producer per broker (unless a different configuration is desired for a certain endpoint)? Wasn't > it the point of our discussion?
With the current endpoint design (if I understand well) the Endpoint.Name == topic name, right? If so, how can we send messages to on different topic? So the basic idea is to create a producer per topic and for each of them define a specific configuration.

Reply:
Yes and no.
That's what I tried to explain you last time. Even though you create multiple KafkaProducer you can make them all share the same Producer  (from the confluentlibrary). The implementation is actually open and depends on the message broker used.